src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,205 @@
+package duke;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Representation of the list containing tasks
+ */
+public class TaskList {
+    private ArrayList<Task> list;
+    private int length;
+    private DateTimeFormatter inputFormatter = DateTimeFormatter.ofPattern("dd-MM-yyyy HH:mm");
+    private DateTimeFormatter storageFormatter = DateTimeFormatter.ofPattern("MMM d yyyy, HH:mm");
+
+    /**
+     * Constructor for the TaskList class
+     * @param tasks
+     */
+    public TaskList(ArrayList<String> tasks) throws DukeException {
+        this.list = new ArrayList<>();
+        for (String s: tasks) {
+            length++;
+            String[] array = s.split("\\|");
+
+            switch (array[0]) {
+            case "T": {
+                this.addTask(new Todo(array[2], Boolean.parseBoolean(array[1])));
+                break;
+            }
+            case "D": {
+                LocalDateTime byDate = LocalDateTime.parse(array[3], storageFormatter);
+                this.addTask(new Deadline(array[2], byDate, Boolean.parseBoolean(array[1])));
+                break;
+            }
+            case "E": {
+                LocalDateTime fromDate = LocalDateTime.parse(array[3], storageFormatter);
+                LocalDateTime toDate = LocalDateTime.parse(array[4], storageFormatter);
+                this.addTask(new Event(array[2], fromDate, toDate, Boolean.parseBoolean(array[1])));
+                break;
+            }
+            default:
+                throw new DukeException("Error Task Type!!");
+            }
+        }
+
+    }
+
+    /**
+     * return size of the array
+     */
+    public int getLength() {
+        return length;
+    }
+
+
+    /**
+     * Adds a task to the list
+     * @param t task to be added to the list
+     */
+    public void addTask(Task t) {
+        list.add(t);
+        length += 1;
+    }
+
+    /**
+     * Checks that the description provided by the user is not empty
+     * @param length
+     * @param error_length
+     * @param task
+     * @throws DukeException
+     */
+    static void validate(Integer length, Integer error_length, String task) throws DukeException {
+        if (length == error_length) {
+            throw new DukeException("☹ OOPS!!! The description of a " + task + " cannot be empty.");
+        }
+    }
+
+    /**
+     * Method that parses the user input
+     * @param userInput input given by the user
+     * @throws DukeException
+     */
+    public void parser(String userInput) throws DukeException {
+        if (userInput.equals("list")) {
```

Comment:
overall, this method seems to be quite long. perhaps it can be split into their own command classes